### DISCUSSION TITLE
Clip/Coerce values of a Vector

### DISCUSSION THREAD

**Discussion by Evangelink** [mathnet/mathnet-numerics#837] [Ideas]
Hi,

I can't seem to find any method to coerce values of a vector between 2 threshold (min and max) I would give. I am not sure if I am not looking from the right word or if no-one ever felt the need for such method.

In case the method doesn't exist but you see value to it, I'd be happy to try and provide a PR for it.

**Comment by cdrnet** [comment]
Hi

Currently we only have two variants of `CoerceZero`, one of them accepting a threshold, the other a predicate function.

But if I understand you correctly, you don't want to coerce to zero but to an interval. With the proposed min-max coerce function, values smaller than min would be replaced with min and values greater than max should be replaced with max?

I wonder how we could extend this to complex vectors.

**Comment by cdrnet** [comment]
You can currently do something like this with the `Map*` functions, but then you have to provide a function that realizes the clipping for a single value.

**Comment by cdrnet** [comment]
Ah, I forgot, we also have the `PointwiseMinimum` and `PointwiseMaximum` functions which are quite like this, but there is no combined variant yet.

**Comment by Evangelink** [comment]
> But if I understand you correctly, you don't want to coerce to zero but to an interval. With the proposed min-max coerce function, values smaller than min would be replaced with min and values greater than max should be replaced with max?

I wasn't thinking about passing the function, rather passing the min and max values. 

> I wonder how we could extend this to complex vectors.

More than happy to provide a PR if we agree on the design and you think that could be valuable for others.
